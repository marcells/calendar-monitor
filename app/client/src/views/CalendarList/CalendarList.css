li {
  list-style-type: none;
}

ul {
  display: table;
  width: 40%;
  padding: 0px;
  margin: 0 auto;
}

a {
  display: block;
  font-size: 200%;
  background: gray;
  border-radius: 5px;
  padding: 15px;
  text-decoration: none;
}

a:link, a:visited {
  color: black;
}

a:hover {
  background: black;
  color: white;
}
