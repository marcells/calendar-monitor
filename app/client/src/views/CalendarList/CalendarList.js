import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'
import { loadCalendars } from '../../redux/actions';

class CalendarList extends Component {
  async componentDidMount() {
    this.props.dispatch(loadCalendars());
  }

  render() {
    return (
      <ul>
        {this.props.items.map(x => <li key={x.id}><Link to={`/calendar/${x.id}`}>{x.description}</Link></li>)}
      </ul>
    );
  }
}

function mapStateToProps(state) {
  const { items } = state.calendars;

  return {
    items
  };
}

export default connect(mapStateToProps)(CalendarList);
