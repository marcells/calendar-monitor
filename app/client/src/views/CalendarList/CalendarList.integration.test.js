import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import moxios from 'moxios';
import CalendarList from './CalendarList';

describe('CalendarList', () => {
  describe('integration', () => {
    beforeEach(() => {
      moxios.install();
    });

    afterEach(() => {
      moxios.uninstall();
    });

    it('should render the calendars correctly', () => {
      const mockStore = configureStore([thunk]);
      const store = mockStore({
        calendars: {
          items: [
            {
              id: 'calendar1',
              description: 'some description for calendar 1'
            },
            {
              id: 'calendar2',
              description: 'some description for calendar 2'
            }
          ]
        }
      });

      const result = renderer.create(
        <Provider store={store}>
          <MemoryRouter>
            <CalendarList />
          </MemoryRouter>
        </Provider>
      );

      expect(result).toMatchSnapshot();
    });
  });
});
