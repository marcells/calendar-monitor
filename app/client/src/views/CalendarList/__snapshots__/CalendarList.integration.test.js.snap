// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarList integration should render the calendars correctly 1`] = `
<ul
  className="CalendarList"
>
  <li>
    <a
      href="/calendar/calendar1"
      onClick={[Function]}
    >
      some description for calendar 1
    </a>
  </li>
  <li>
    <a
      href="/calendar/calendar2"
      onClick={[Function]}
    >
      some description for calendar 2
    </a>
  </li>
</ul>
`;
